stl/inc/xstring

```diff
@@ -2432,6 +2438,33 @@ public:
         _Proxy._Release();
     }
 
+    explicit basic_string(_String_constructor_concat_tag, const basic_string& _Alsource, const _Elem* const _Left_ptr,
+        const size_type _Left_size, const _Elem* const _Right_ptr, const size_type _Right_size)
+        : _Mypair(
+            _One_then_variadic_args_t(), _Alty_traits::select_on_container_copy_construction(_Alsource._Getal())) {
+        _STL_INTERNAL_CHECK(_Left_size < max_size());
+        _STL_INTERNAL_CHECK(_Right_size < max_size());
+        _STL_INTERNAL_CHECK(max_size() - _Left_size >= _Right_size);
+        const auto _New_size    = _Left_size + _Right_size;
+        size_type _New_capacity = _BUF_SIZE - 1;
+        _Elem* _Ptr             = _Mypair._Myval2._Bx._Buf;

```

Comment:
You repeat `_Mypair._Myval2` several times; I believe we usually factor this out into `_My_data` which is zero-overhead.